Fix warnings

robamler opened the pull request:
While working on #25 I fund it a bit tedious to debug changes because error messages were buried in quite a lot of warnings from existing code. In particular, this made it difficult to see at a glance whether the newly contributed code raised any warnings.

This PR suggests to fix all warnings in existing code. This is not meant as a complaint, I'm very grateful that this crate exists. I just think that having no existing warnings reduces mental burden on external contributors.

The PR is split into two commits: the first one fixes all compiler warnings and the second one fixes all clippy warnings. All fixes turned out to be trivial, so there should be no change in run-time behavior or performance.

IvanUkhov commented on src/distribution/binomial.rs:
```diff
@@ -353,21 +348,21 @@ fn stirlerr(n: f64) -> f64 {
     // See [Loader, 2000, pp. 7].
     const SFE: [f64; 16] = [
         0.000000000000000000e+00,
-        8.106146679532725822e-02,
```
Please keep it as it was.

IvanUkhov commented on src/distribution/binomial.rs:
```diff
@@ -319,26 +317,23 @@ fn approximate_by_normal(p: f64, np: f64, v: f64, u: f64) -> f64 {
     let p3 = p2 * p;
     let p4 = p2 * p2;
 
-    np +
```
Please keep it as it was.

IvanUkhov commented on src/distribution/gaussian.rs:
```diff
@@ -156,77 +156,77 @@ impl distribution::Variance for Gaussian {
     const SPLIT1: f64 = 0.425;
     const SPLIT2: f64 = 5.0;
     const A: [f64; 8] = [
-        3.3871328727963666080e+00,
```
Please keep it as it was, including B, C, D, E, and F.

IvanUkhov commented on src/distribution/gaussian.rs:
```diff
@@ -156,77 +156,77 @@ impl distribution::Variance for Gaussian {
 
     #[inline(always)]
     fn poly(c: &[f64], x: f64) -> f64 {
-        c[0] + x * (c[1] + x * (c[2] + x * (c[3] + x * (
```
Please keep it as it was.

IvanUkhov commented on src/distribution/gaussian.rs:
```diff
@@ -295,108 +295,283 @@ pub fn sample<S: Source>(source: &mut S) -> f64 {
 const R: f64 = 3.44428647676;
 
 const K: [u32; 128] = [
-    00000000, 12590644, 14272653, 14988939,
```
Please keep it as it was, including Y and W.

IvanUkhov reviewed (changes requested):
Thank you for the PR! Yes, we should certainly eliminate all the warnings. Please undo all changes related to constants and formatting.

robamler commented:
Thanks for the quick response! I reverted all parts that you asked for and added compiler directives to prevent such changes to happen accidentally in the future (e.g., when other contributors have their editors set to auto format on save).

Should I squash the last two commits?

IvanUkhov commented on src/distribution/binomial.rs:
```diff
@@ -242,9 +242,7 @@ impl distribution::Median for Binomial {
-        if self.np.fract() == 0.0 {
-            self.np
-        } else if self.p == 0.5 && self.n % 2 != 0 {
+        if self.np.fract() == 0.0 || (self.p == 0.5 && self.n % 2 != 0) {
```
Why is this change needed?

robamler replied:
I don't think it's necessary, but `rustc` raised a warning that both the `if` and the first `else if` branch had identical bodies, so they could be collapsed into a single branch.

robamler replied:
I guess the warning is mostly meant to catch copy+paste mistakes where the bodies should actually be different.

IvanUkhov replied:
I see. Then it is justified.

IvanUkhov commented on src/distribution/binomial.rs:
```diff
@@ -301,6 +299,7 @@ impl distribution::Variance for Binomial {
 }
 
 // See [Moorhead, 2013, pp. 7].
+#[rustfmt::skip]
```
Nice! Thank you.

IvanUkhov commented on src/distribution/binomial.rs:
```diff
@@ -351,6 +350,7 @@ fn stirlerr(n: f64) -> f64 {
     const S4: f64 = 1.0 / 1188.0;
 
     // See [Loader, 2000, pp. 7].
+    #[allow(clippy::excessive_precision)]
```
Aha, it was clippy complaining. Nice to have this annotation!

IvanUkhov commented on src/distribution/gaussian.rs:
```diff
@@ -225,8 +226,8 @@ pub fn inverse(p: f64) -> f64 {
-        c[0] + x * (c[1] + x * (c[2] + x * (c[3] + x * (
-        c[4] + x * (c[5] + x * (c[6] + x * (c[7])))))))
+        c[0] + x
+            * (c[1] + x * (c[2] + x * (c[3] + x * (c[4] + x * (c[5] + x * (c[6] + x * (c[7])))))))
```
Please annotation this function too.

robamler replied:
Done.

IvanUkhov commented:
Thank you!
